twill-yarn/src/main/java/org/apache/twill/internal/appmaster/ApplicationMasterService.java

```diff
@@ -135,6 +135,9 @@
   private final Map<String, Map<String, String>> environments;
   private final TwillRuntimeSpecification twillRuntimeSpec;
 
+  private volatile boolean completed;
```

Comment:
Instead of having three variables for `completed`, `aborted`, `stopped`, it's better to have one Enum to represent them.